repository_service_tuf/cli/admin/ceremony.py

```diff
@@ -229,219 +238,439 @@
 console = Console()
 
 
-class Roles(Enum):
-    ROOT = "root"
-    TARGETS = "targets"
-    SNAPSHOT = "snapshot"
-    TIMESTAMP = "timestamp"
-    BINS = "bins"
-
+# Define all initial settings
+setup = Setup(
+    expiration={
+        Roles.ROOT: 365,
+        Roles.TARGETS: 365,
+        Roles.SNAPSHOT: 1,
+        Roles.TIMESTAMP: 1,
+        Roles.BINS: 1,
+    },
+    services=ServiceSettings(),
+    number_of_keys={Roles.ROOT: 2, Roles.TARGETS: 1},
+    threshold={
+        Roles.ROOT: 1,
+        Roles.TARGETS: 1,
+    },
+    keys={
+        Roles.ROOT: [],
+        Roles.TARGETS: [],
+        Roles.SNAPSHOT: [],
+        Roles.TIMESTAMP: [],
+        Roles.BINS: [],
+    },
+    online_key=RSTUFKey(),
+)
 
-@dataclass
-class RoleSettings:
-    expiration: int
-    threshold: int
-    keys: int
-    offline_keys: bool
 
+def _key_is_duplicated(key: Dict[str, Any]) -> bool:
+    """Check if a Key is duplicated, used in a role or online_key"""
+    # verify if key is in any Roke keys
+    for role_key in setup.keys.values():
+        if any(
+            k
+            for k in role_key
```

Comment:
No, `setup.keys` is a `Dict[Literal[Roles], List[RSTUFKey]]`
It interacts with all the keys in the role names. In other words, Root and online keys as well.

BTW: It would be another change required by your [suggestion comment here](https://github.com/vmware/repository-service-tuf-cli/pull/201#discussion_r1121705588)

Reply:
I made it more explicitly on my latest commit renaming it from `role_key` to `role_keys`

Reply:
Ookay, now when you renamed it's better.
I read it incorrectly.